lib/shopify-cli/project_type.rb

```diff
@@ -54,13 +58,15 @@ def hidden_project_type
       end
 
       def register_command(const, cmd)
+        return if project_load_shallow
         Context.new.abort(
           Context.message('core.project_type.error.cannot_override_core', cmd, const)
         ) if Commands.core_command?(cmd)
         Commands.register(const, cmd)
       end
 
       def register_task(task, name)
+        return if project_load_shallow
         Task::Registry.add(const_get(task), name)
       end
 
```

Comment:
Should we also not load the messages if we're shallow loading? That would reduce the overhead of reading the files.